airbyte-commons-worker/src/main/java/io/airbyte/workers/test_utils/AirbyteMessageUtils.java

```diff
@@ -102,29 +103,43 @@ public static AirbyteStreamState createStreamState(final String streamName) {
     return new AirbyteStreamState().withStreamDescriptor(new StreamDescriptor().withName(streamName));
   }
 
+  public static AirbyteMessage createEstimateMessage(final String namespace, final String name, final long byteEst, final long rowEst) {
+    final var est = new AirbyteEstimateTraceMessage()
+        .withType(AirbyteEstimateTraceMessage.Type.STREAM)
+        .withNamespace(namespace)
+        .withName(name)
+        .withByteEstimate(byteEst)
+        .withRowEstimate(rowEst);
+
+    return new AirbyteMessage()
+        .withType(Type.TRACE)
+        .withTrace(new AirbyteTraceMessage().withType(AirbyteTraceMessage.Type.ESTIMATE)
+            .withEstimate(est));
+  }
+
+  public static AirbyteMessage createErrorMessage(final String message, final Double emittedAt) {
+    return new AirbyteMessage()
+        .withType(AirbyteMessage.Type.TRACE)
+        .withTrace(createErrorTraceMessage(message, emittedAt));
+  }
+
   public static AirbyteTraceMessage createErrorTraceMessage(final String message, final Double emittedAt) {
-    return new AirbyteTraceMessage()
-        .withType(io.airbyte.protocol.models.AirbyteTraceMessage.Type.ERROR)
-        .withEmittedAt(emittedAt)
-        .withError(new AirbyteErrorTraceMessage().withMessage(message));
+    return createErrorTraceMessage(message, emittedAt, null);
```

Comment:
consolidate the various trace message creation into one function to avoid duplication.